From f9e9c2f3bebb32cdd45646ba85c19e7839e99f9b Mon Sep 17 00:00:00 2001
From: "Radivoje (Ogi) Jovanovic" <rjovanovic@nvidia.com>
Date: Sat, 3 Sep 2022 15:19:33 -0700
Subject: [PATCH] driver: i2c: mux: pca954x: enable support for numbered
 adapters

there is a need to control the bus numbers taken by pca 954x in some
configurations. Enable the support over dts

Signed-off-by: Radivoje (Ogi) Jovanovic <rjovanovic@nvidia.com>
Fixes JIRA https://jirasw.nvidia.com/browse/DGXOPENBMC-5353
Upstream-Status: Pending [Not submitted to upstream yet]
---
 drivers/i2c/muxes/i2c-mux-pca954x.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/i2c/muxes/i2c-mux-pca954x.c b/drivers/i2c/muxes/i2c-mux-pca954x.c
index 6f84018258c4..8a73429f45d2 100644
--- a/drivers/i2c/muxes/i2c-mux-pca954x.c
+++ b/drivers/i2c/muxes/i2c-mux-pca954x.c
@@ -561,8 +561,10 @@ static int pca954x_probe(struct i2c_client *client)
 	struct device *dev = &client->dev;
 	struct i2c_mux_core *muxc;
 	struct pca954x *data;
+	bool nr_taken = true;
 	int num;
 	int ret;
+	int nr;
 
 	if (!i2c_check_functionality(adap, I2C_FUNC_SMBUS_BYTE))
 		return -ENODEV;
@@ -642,9 +644,16 @@ static int pca954x_probe(struct i2c_client *client)
 	if (ret)
 		goto fail_cleanup;
 
+	if (device_property_read_u32(dev, "numbered-adapter", &nr)) {
+		nr = 0;
+		nr_taken = false;
+	}
 	/* Now create an adapter for each channel */
 	for (num = 0; num < data->chip->nchans; num++) {
-		ret = i2c_mux_add_adapter(muxc, 0, num);
+		int n = 0;
+		if (nr_taken)
+			n = nr + num;
+		ret = i2c_mux_add_adapter(muxc, n, num);
 		if (ret)
 			goto fail_cleanup;
 	}
-- 
2.43.0

